Quaver.Shared/Screens/Select/SelectScreen.cs

```diff
@@ -542,5 +573,184 @@ public void SelectRandomMap()
                     break;
             }
         }
+
+        /// <summary>
+        ///     Used to start the deleting process.
+        ///
+        ///     woo now AiAe will stop telling me to add this.
+        /// </summary>
+        /// <param name="type"></param>
+        private void DeleteSelected()
+        {
+            var view = View as SelectScreenView;
+            var type = view.ActiveContainer;
+            var filePath = string.Empty;
+
+            var selectedMapset = AvailableMapsets[view.MapsetScrollContainer.SelectedMapsetIndex];
+
+            if (AvailableMapsets.Count == 0)
+                return;
+
+            switch (type)
+            {
+                case SelectContainerStatus.Mapsets:
+                    if (selectedMapset == null)
+                        return;
+
+                    filePath = Path.Combine(ConfigManager.SongDirectory.Value, selectedMapset.Directory);
+                    break;
+
+                case SelectContainerStatus.Difficulty:
+                    var selectedDifficulty = AvailableMapsets[view.MapsetScrollContainer.SelectedMapsetIndex].Maps[view.DifficultyScrollContainer.SelectedMapIndex];
+
+                    if (selectedDifficulty == null)
+                        return;
+
+                    filePath = selectedMapset.Maps.Count > 1 ? Path.Combine(Path.Combine(ConfigManager.SongDirectory.Value, selectedMapset.Directory), selectedDifficulty.Path) : Path.Combine(ConfigManager.SongDirectory.Value, selectedMapset.Directory);
```

Comment:
Extract `Path.Combine(ConfigManager.SongDirectory.Value, selectedMapset.Directory)` into a variable right before this statement for readability.